js/modules/loki_message_api.js

```diff
@@ -40,79 +90,53 @@ class LokiMessageAPI {
     let successfulRequests = 0;
     let canResolve = true;
 
-    const doRequest = async nodeUrl => {
-      // TODO: Confirm sensible timeout
-      const options = {
-        url: `${nodeUrl}${this.messageServerPort}/store`,
-        type: 'POST',
-        responseType: undefined,
-        timeout: 10000,
-      };
+    let swarmNodes = await window.Signal.Data.getSwarmNodesByPubkey(pubKey);
 
+    const doRequest = async nodeUrl => {
+      const url = `${nodeUrl}${this.messageServerPort}/store`;
       const fetchOptions = {
-        method: options.type,
+        method: 'POST',
         body: data64,
         headers: {
           'X-Loki-pow-nonce': nonce,
           'X-Loki-timestamp': timestamp.toString(),
           'X-Loki-ttl': ttl.toString(),
           'X-Loki-recipient': pubKey,
         },
-        timeout: options.timeout,
       };
 
-      let response;
       try {
-        response = await fetch(options.url, fetchOptions);
-      } catch (e) {
-        if (e.code === 'ENOTFOUND') {
-          // TODO: Handle the case where lokinet is not working
-          canResolve = false;
-          return;
-        }
-        log.error(options.type, options.url, 0, 'Error sending message');
-        if (window.LokiSnodeAPI.unreachableNode(pubKey, nodeUrl)) {
-          completedNodes.push(nodeUrl);
-          swarmNodes = swarmNodes.filter(node => node !== nodeUrl);
-        }
-        return;
-      }
-
-      let result;
-      if (
-        options.responseType === 'json' &&
-        response.headers.get('Content-Type') === 'application/json'
-      ) {
-        result = await response.json();
-      } else if (options.responseType === 'arraybuffer') {
-        result = await response.buffer();
-      } else {
-        result = await response.text();
-      }
+        await fetch(url, fetchOptions);
 
-      if (response.status >= 0 && response.status < 400) {
         completedNodes.push(nodeUrl);
         swarmNodes = swarmNodes.filter(node => node !== nodeUrl);
         successfulRequests += 1;
-        return;
+      } catch (e) {
+        if (e instanceof NotFoundError) {
+          canResolve = false;
+        } else if (e instanceof HTTPError) {
+          log.error(
```

Comment:
Also I think I just realised that we should be marking the node as complete if we get a HTTPError because otherwise we would just keep querying them and getting rejected? I think that was missed in my logic